force-app/main/default/lwc/apexImperativeMethodWithComplexParams/__tests__/apexImperativeMethodWithComplexParams.test.js

```diff
@@ -0,0 +1,184 @@
+            {
+                is: ApexImperativeMethodWithComplexParams
+            }
+        );
+        document.body.appendChild(element);
+
+        // Select input field for simulating string user input
+        const inputStringEl = element.shadowRoot.querySelector(
+            'lightning-input[class="string-input"]'
+        );
+        inputStringEl.value = APEX_PARAMETER.someString;
+        inputStringEl.dispatchEvent(new CustomEvent('change'));
+
+        // Select input field for simulating number user input
+        const inputNumberEl = element.shadowRoot.querySelector(
+            'lightning-input[class="number-input"]'
+        );
+        inputNumberEl.value = APEX_PARAMETER.someInteger;
+        inputNumberEl.dispatchEvent(new CustomEvent('change'));
+
+        // Select input field for simulating list item user input
+        const inputListItemEl = element.shadowRoot.querySelector(
+            'lightning-input[class="list-item-input"]'
+        );
+        inputListItemEl.value = APEX_PARAMETER.someList.length;
+        inputListItemEl.dispatchEvent(new CustomEvent('change'));
+
+        // Select button for executing Apex call
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return an immediate flushed promise (after the Apex call) to then
+        // wait for any asynchronous DOM updates. Jest will automatically wait
+        // for the Promise chain to complete before ending the test and fail
+        // the test if the promise ends in the rejected state.
+        return flushPromises().then(() => {
+            // Select p for validating conditionally changed text content
+            const detailEl = element.shadowRoot.querySelector('p');
+            expect(detailEl.textContent).toBe(APEX_SUCCESS);
+        });
+    });
+
+    it('renders the error panel when the Apex method returns an error', () => {
+        // Assing mock value for rejected Apex promise
+        checkApexTypes.mockRejectedValue(APEX_ERROR);
+
+        // Create initial element
+        const element = createElement('c-apex-imperative-method-with-params', {
+            is: ApexImperativeMethodWithComplexParams
+        });
+        document.body.appendChild(element);
+
+        // Select button for executing Apex call
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return an immediate flushed promise (after the Apex call) to then
+        // wait for any asynchronous DOM updates. Jest will automatically wait
+        // for the Promise chain to complete before ending the test and fail
+        // the test if the promise ends in the rejected state.
+        return flushPromises().then(() => {
+            const errorPanelEl = element.shadowRoot.querySelector(
+                'c-error-panel'
+            );
+            expect(errorPanelEl).not.toBeNull();
```

Comment:
Consider checking detailEl is empty in this case. 

And checking the inverse (error label is null) in a success case.